hi-ml-azure/testazure/testazure/test_himl.py

```diff
@@ -1706,6 +1714,67 @@ def test_get_display_name_v2() -> None:
     assert display_name == ""
 
 
+def test_submitting_script_with_sdk_v2(tmp_path: Path) -> None:
+    """
+    Test that submitting a simple script can be run on AzureML when using the v2 SDK.
+    It also tests the "wait_for_completion" parameter.
+    """
+    # Create a minimal script in a temp folder. Script
+    test_script = tmp_path / "test_script.py"
+    test_script.write_text("print('hello world')")
+    shared_config_json = get_shared_config_json()
+    conda_env_path = create_empty_conda_env(tmp_path)
+    after_submission_called = False
+
+    def after_submission(job: Job, ml_client: MLClient) -> None:
+        nonlocal after_submission_called
+        after_submission_called = True
+        assert isinstance(job, Job)
+        assert isinstance(ml_client, MLClient)
+        # We waited for completion of the job, so it's status should now be "Completed"
+        assert job.status == JobStatus.Completed.value
+
+    with check_config_json(tmp_path, shared_config_json=shared_config_json),\
```

Comment:
For completeness could we also check that when wait_for_completion=False, after_submission does not get called?